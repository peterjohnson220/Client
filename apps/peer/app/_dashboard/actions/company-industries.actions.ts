import { Action } from '@ngrx/store';

export const LOAD_COMPANY_INDUSTRIES  = '[Peer Main/Dashboard/Request Company/Company Industries] Load Company Industries';
export const LOAD_COMPANY_INDUSTRIES_SUCCESS  = '[Peer Main/Dashboard/Request Company/Company Industries] Load Company Industries Success';
export const LOAD_COMPANY_INDUSTRIES_ERROR  = '[Peer Main/Dashboard/Request Company/Company Industries] Load Company Industries Error';

export class LoadCompanyIndustries implements Action {
  readonly type = LOAD_COMPANY_INDUSTRIES;
}

export class LoadCompanyIndustriesSuccess implements Action {
  readonly type = LOAD_COMPANY_INDUSTRIES_SUCCESS;

  constructor(public payload: string[]) {}
}

export class LoadCompanyIndustriesError implements Action {
  readonly type = LOAD_COMPANY_INDUSTRIES_ERROR;
}

export type Actions
  = LoadCompanyIndustries
  | LoadCompanyIndustriesSuccess
  | LoadCompanyIndustriesError;
