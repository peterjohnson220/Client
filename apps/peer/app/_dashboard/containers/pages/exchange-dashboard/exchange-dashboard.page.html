<pf-page-with-title-bar>

  <ng-container page-title>
    <pf-peer-dashboard-exchange-selector></pf-peer-dashboard-exchange-selector>
  </ng-container>

  <ng-container page-actions>
    <button type="button" class="btn btn-secondary" [attr.data-qa-id]="'btn-inviteCompany'" (click)="referCompanyClick()">Invite Company</button>
    <button type="button" class="btn btn-secondary" [attr.data-qa-id]="'btn-manageJobs'" (click)="manageJobsClick()">Manage Jobs</button>
    <button type="button" class="btn btn-primary" [attr.data-qa-id]="'btn-manageScopesAndExport'" (click)="mapClick()" [disabled]="!(mapHasData$ | async)"
    [title]="getTitle((mapHasData$ | async), (mapHasDataError$ | async))">Manage Scopes & Export</button>
  </ng-container>

  <ng-container page-content>
    <div class="exchange-dashboard-container d-flex flex-row">
      <div class="exchange-dashboard-content d-flex flex-column">
        <div class="exchange-dashboard-chart-container d-flex flex-row justify-content-around">
          <div class="d-flex flex-column justify-content-between mr-2 count-column">
            <div class="d-flex card card-body flex-column flex-grow-1 text-center mb-1 justify-content-center">
              <pf-exchange-company-count></pf-exchange-company-count>
            </div>
            <div class="d-flex card card-body flex-column flex-grow-1 text-center mt-1 justify-content-center">
              <pf-exchange-job-count></pf-exchange-job-count>
            </div>
          </div>
          <div class="d-flex flex-grow-2 flex-column justify-content-between mr-2">
            <div class="card card-body flex-column justify-content-top flex-grow-1 text-center">
              <h4>Top Industries</h4>
              <div class="d-flex flex-grow-1 flex-column justify-content-center text-center">
                <pf-exchange-industry-chart></pf-exchange-industry-chart>
              </div>
            </div>
          </div>
          <div class="d-flex flex-grow-2 flex-column justify-content-between mr-2">
            <div class="card card-body flex-column justify-content-top flex-grow-1 text-center">
              <h4>Top Exchange Job Families</h4>
              <div class="d-flex flex-grow-1 flex-column justify-content-center text-center">
                <div>
                  <pf-exchange-job-family-chart></pf-exchange-job-family-chart>
                </div>
              </div>
            </div>
          </div>
          <div class="d-flex flex-grow-2 flex-column justify-content-between">
            <div class="card card-body flex-column justify-content-top flex-grow-1 text-center">
              <h4>Revenue Size</h4>
              <div class="d-flex flex-grow-1 flex-column justify-content-center text-center">
                <div>
                  <pf-exchange-revenue-chart></pf-exchange-revenue-chart>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="exchange-dashboard-job-comparison-grid mt-3 d-flex flex-column">
          <div class="h-100">
            <pf-exchange-job-comparison-grid></pf-exchange-job-comparison-grid>
          </div>
        </div>
      </div>
      <div class="exchange-dashboard-sidebar" *ngIf="(sidebarVisible$ | async)">
        <pf-chart-detail-component></pf-chart-detail-component>
      </div>
    </div>
  </ng-container>

</pf-page-with-title-bar>

<pf-invite-company-modal></pf-invite-company-modal>
<pf-access-exchange-request-modal></pf-access-exchange-request-modal>
<pf-exchange-dashboard-tc-modal></pf-exchange-dashboard-tc-modal>
<pf-upload-org-data-modal
  [isOpen$]="uploadOrgDataModalOpen$"
  (uploadOrgDataEvent)="handleUploadOrgData($event)"
  (modalDismissedEvent)="closeUploadOrgDataModal()">
</pf-upload-org-data-modal>
