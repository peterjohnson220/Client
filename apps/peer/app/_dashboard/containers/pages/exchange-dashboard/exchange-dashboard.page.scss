.exchange-dashboard-container {
  &, .exchange-dashboard-chart-container, .exchange-dashboard-job-comparison-grid {
    height: 100%;
  }

  .exchange-dashboard-job-comparison-grid {
    position: relative;
  }

  .exchange-dashboard-content {
    flex: 1 0;
  }

  pf-report-preview {
    overflow: hidden;
    max-width: fit-content;
    min-width: 300px;
  }
}

.exchange-dashboard-sidebar {
  min-width: 300px;
  max-width: 300px;
}

.count-column {
  min-width: 200px;
}

.exchange-job-count {
  cursor: pointer;
}
