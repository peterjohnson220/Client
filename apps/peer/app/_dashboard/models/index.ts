export * from './existing-company.model';
export * from './exchanges-and-last-visted.model';
