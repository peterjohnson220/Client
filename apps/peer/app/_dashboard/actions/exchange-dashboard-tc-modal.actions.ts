import { Action } from '@ngrx/store';

import { TermsConditionsModel, TermsConditionsSubmissionModel } from 'libs/models';

export const LOAD_TC =  '[Peer Dashboard/TC Modal] Load Terms and Conditions';
export const LOAD_TC_SUCCESS =  '[Peer Dashboard/TC Modal] Load Terms and Conditions Success';
export const LOAD_TC_ERROR =  '[Peer Dashboard/TC Modal] Load Terms and Conditions Error';
export const SUBMIT_TC = '[Peer Dashboard/TC Modal] Submit Terms and Conditions Response';
export const SUBMIT_TC_SUCCESS = '[Peer Dashboard/TC Modal] Submit Terms and Conditions Response Success';
export const SUBMIT_TC_ERROR = '[Peer Dashboard/TC Modal] Submit Terms and Conditions Response Error';

export class LoadTermsAndConditions implements Action {
  readonly type = LOAD_TC;
  constructor(public payload: string) {}
}

export class LoadTermsAndConditionsSuccess implements Action {
  readonly type = LOAD_TC_SUCCESS;

  constructor(public payload: TermsConditionsModel) {}
}

export class LoadTermsAndConditionsError implements Action {
  readonly type = LOAD_TC_ERROR;

  constructor(public any: Error) {}
}

export class SubmitTermsAndConditionsResponse implements Action {
  readonly type = SUBMIT_TC;

  constructor(public payload: TermsConditionsSubmissionModel) {}
}

export class SubmitTermsAndConditionsResponseSuccess implements Action {
  readonly type = SUBMIT_TC_SUCCESS;

  constructor() {}
}

export class SubmitTermsAndConditionsResponseError implements Action {
  readonly type = SUBMIT_TC_ERROR;

  constructor(public payload: TermsConditionsModel) {}
}


export type Actions
  = LoadTermsAndConditions
  | LoadTermsAndConditionsSuccess
  | LoadTermsAndConditionsError
  | SubmitTermsAndConditionsResponse
  | SubmitTermsAndConditionsResponseSuccess
  | SubmitTermsAndConditionsResponseError;
