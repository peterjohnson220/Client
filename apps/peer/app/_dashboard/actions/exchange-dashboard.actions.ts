import { Action } from '@ngrx/store';

import { GetChartRequest, GetDetailChartRequest, ChartItem } from 'libs/models';

export const LOADING_COMPANY_CHART  = '[Peer Dashboard/Page] Loading Company Chart';
export const LOADING_COMPANY_CHART_SUCCESS  = '[Peer Dashboard/Page] Loading Company Chart Success';
export const LOADING_COMPANY_CHART_ERROR  = '[Peer Dashboard/Page] Loading Company Chart Error';
export const LOADING_JOB_CHART  = '[Peer Dashboard/Page] Loading Job Chart';
export const LOADING_JOB_CHART_SUCCESS  = '[Peer Dashboard/Page] Loading Job Chart Success';
export const LOADING_JOB_CHART_ERROR  = '[Peer Dashboard/Page] Loading Job Chart Error';
export const LOADING_INDUSTRY_CHART  = '[Peer Dashboard/Page] Loading Industry Chart';
export const LOADING_INDUSTRY_CHART_SUCCESS  = '[Peer Dashboard/Page] Loading Industry Chart Success';
export const LOADING_INDUSTRY_CHART_ERROR  = '[Peer Dashboard/Page] Loading Industry Chart Error';
export const LOADING_JOB_FAMILY_CHART  = '[Peer Dashboard/Page] Loading Job Family Chart';
export const LOADING_JOB_FAMILY_CHART_SUCCESS  = '[Peer Dashboard/Page] Loading Job Family Chart Success';
export const LOADING_JOB_FAMILY_CHART_ERROR  = '[Peer Dashboard/Page] Loading Job Family Chart Error';
export const LOADING_REVENUE_CHART  = '[Peer Dashboard/Page] Loading Revenue Chart';
export const LOADING_REVENUE_CHART_SUCCESS  = '[Peer Dashboard/Page] Loading Revenue Chart Success';
export const LOADING_REVENUE_CHART_ERROR  = '[Peer Dashboard/Page] Loading Revenue Chart Error';
export const LOADING_DETAIL_CHART  = '[Peer Dashboard/Page] Loading Detail Chart';
export const LOADING_DETAIL_CHART_SUCCESS  = '[Peer Dashboard/Page] Loading Detail Chart Success';
export const LOADING_DETAIL_CHART_ERROR  = '[Peer Dashboard/Page] Loading Detail Chart Error';
export const CLOSE_SIDEBAR  = '[Peer Dashboard/Page] Close Sidebar';

export class LoadingCompanyChart implements Action {
  readonly type = LOADING_COMPANY_CHART;

  constructor(public payload: GetChartRequest) {}
}

export class LoadingCompanyChartSuccess implements Action {
  readonly type = LOADING_COMPANY_CHART_SUCCESS;

  constructor(public payload: ChartItem[]) {}
}

export class LoadingCompanyChartError implements Action {
  readonly type = LOADING_COMPANY_CHART_ERROR;
}

export class LoadingJobChart implements Action {
  readonly type = LOADING_JOB_CHART;

  constructor(public payload: GetChartRequest) {}
}

export class LoadingJobChartSuccess implements Action {
  readonly type = LOADING_JOB_CHART_SUCCESS;

  constructor(public payload: ChartItem[]) {}
}

export class LoadingJobChartError implements Action {
  readonly type = LOADING_JOB_CHART_ERROR;
}

export class LoadingIndustryChart implements Action {
  readonly type = LOADING_INDUSTRY_CHART;

  constructor(public payload: GetChartRequest) {}
}

export class LoadingIndustryChartSuccess implements Action {
  readonly type = LOADING_INDUSTRY_CHART_SUCCESS;

  constructor(public payload: ChartItem[]) {}
}

export class LoadingIndustryChartError implements Action {
  readonly type = LOADING_INDUSTRY_CHART_ERROR;
}

export class LoadingJobFamilyChart implements Action {
  readonly type = LOADING_JOB_FAMILY_CHART;

  constructor(public payload: GetChartRequest) {}
}

export class LoadingJobFamilyChartSuccess implements Action {
  readonly type = LOADING_JOB_FAMILY_CHART_SUCCESS;

  constructor(public payload: ChartItem[]) {}
}

export class LoadingJobFamilyChartError implements Action {
  readonly type = LOADING_JOB_FAMILY_CHART_ERROR;
}

export class LoadingRevenueChart implements Action {
  readonly type = LOADING_REVENUE_CHART;

  constructor(public payload: GetChartRequest) {}
}

export class LoadingRevenueChartSuccess implements Action {
  readonly type = LOADING_REVENUE_CHART_SUCCESS;

  constructor(public payload: ChartItem[]) {}
}

export class LoadingRevenueChartError implements Action {
  readonly type = LOADING_REVENUE_CHART_ERROR;
}

export class LoadingDetailChart implements Action {
  readonly type = LOADING_DETAIL_CHART;

  constructor(public payload: GetDetailChartRequest) {}
}

export class LoadingDetailChartSuccess implements Action {
  readonly type = LOADING_DETAIL_CHART_SUCCESS;

  constructor(public payload: ChartItem[]) {}
}

export class LoadingDetailChartError implements Action {
  readonly type = LOADING_DETAIL_CHART_ERROR;
}

export class CloseSidebar implements Action {
  readonly type = CLOSE_SIDEBAR;
}

export type Actions
  = LoadingCompanyChart
  | LoadingCompanyChartSuccess
  | LoadingCompanyChartError
  | LoadingJobChart
  | LoadingJobChartSuccess
  | LoadingJobChartError
  | LoadingIndustryChart
  | LoadingIndustryChartSuccess
  | LoadingIndustryChartError
  | LoadingJobFamilyChart
  | LoadingJobFamilyChartSuccess
  | LoadingJobFamilyChartError
  | LoadingRevenueChart
  | LoadingRevenueChartSuccess
  | LoadingRevenueChartError
  | LoadingDetailChart
  | LoadingDetailChartSuccess
  | LoadingDetailChartError
  | CloseSidebar;
