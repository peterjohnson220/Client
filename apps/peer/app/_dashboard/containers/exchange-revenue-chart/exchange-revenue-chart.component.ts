import { Component } from '@angular/core';
import { ActivatedRoute, ParamMap } from '@angular/router';

import { Store } from '@ngrx/store';
import { Observable } from 'rxjs/Observable';

import { GetDetailChartRequest, ChartItem, ExchangeChartTypeEnum } from 'libs/models';

import * as fromExchangeDashboardActions from '../../actions/exchange-dashboard.actions';
import * as fromPeerDashboardReducer from '../../reducers';

@Component({
  selector: 'pf-exchange-revenue-chart',
  templateUrl: './exchange-revenue-chart.component.html',
  styleUrls: ['./exchange-revenue-chart.component.scss']
})

export class ExchangeRevenueChartComponent {
  exchangeId: number;
  revenueChartItems$: Observable<ChartItem[]>;
  loadingRevenueChartItems$: Observable<boolean>;
  loadingRevenueChartItemsError$: Observable<boolean>;

  constructor(
    private store: Store<fromPeerDashboardReducer.State>,
    private route: ActivatedRoute
  ) {
    this.revenueChartItems$ = this.store.select(fromPeerDashboardReducer.getExchangeDashboardRevenueChartItems);
    this.loadingRevenueChartItems$ = this.store.select(fromPeerDashboardReducer.getExchangeDashboardLoadingRevenueChart);
    this.loadingRevenueChartItemsError$ = this.store.select(fromPeerDashboardReducer.getExchangeDashboardLoadingRevenueChartError);
    this.route.paramMap.subscribe((params: ParamMap) => this.exchangeId = +params.get('id'));
  }

  seriesClick(e): void {
    const getDetailChartRequest: GetDetailChartRequest = {
      ExchangeId: this.exchangeId,
      ChartType: ExchangeChartTypeEnum.Revenue,
      Category: e.category
    };
    this.store.dispatch(new fromExchangeDashboardActions.LoadingDetailChart(getDetailChartRequest));
  }
}
