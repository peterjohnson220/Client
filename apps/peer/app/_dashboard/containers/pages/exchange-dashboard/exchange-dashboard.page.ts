import { Component, OnInit, OnDestroy, ViewChild } from '@angular/core';
import { ActivatedRoute, Router } from '@angular/router';

import { Store, select } from '@ngrx/store';
import { Observable, Subject, Subscription } from 'rxjs';

import { ExchangeRequestTypeEnum, Exchange, AsyncStateObj } from 'libs/models';
import * as fromCompanyContextActions from 'libs/state/app-context/actions/company-context.actions';
import { Permissions } from 'libs/constants';
import { Workbook } from 'libs/features/surveys/reports/models';
import { AbstractFeatureFlagService, FeatureFlags, RealTimeFlag } from 'libs/core/services/feature-flags';
import { SettingsService } from 'libs/state/app-context/services';
import { FileDownloadSecurityWarningModalComponent } from 'libs/ui/common';
import { CompanySettingsEnum } from 'libs/models/company';

import * as fromExchangeDashboardActions from '../../../actions/exchange-dashboard.actions';
import * as fromUploadOrgDataAction from '../../../actions/upload-org-data.actions';
import * as fromExchangeRequestActions from '../../../../shared/actions/exchange-request.actions';
import * as fromPeerDashboardReducer from '../../../reducers';
import * as fromSharedPeerReducer from '../../../../shared/reducers';
import { PeerTrendReportModalComponent } from '../../../components/peer-trend-report-modal';

@Component({
  selector: 'pf-exchange-dashboard-page',
  templateUrl: './exchange-dashboard.page.html',
  styleUrls: ['./exchange-dashboard.page.scss']
})
export class ExchangeDashboardPageComponent implements OnInit, OnDestroy {
  @ViewChild(PeerTrendReportModalComponent) peerTrendReportModal: PeerTrendReportModalComponent;
  @ViewChild('fileDownloadSecurityWarningModal', { static: true }) fileDownloadSecurityWarningModal: FileDownloadSecurityWarningModalComponent;

  sidebarVisible$: Observable<boolean>;
  mapHasData$: Observable<boolean>;
  mapHasDataError$: Observable<boolean>;
  uploadOrgDataModalOpen$: Observable<boolean>;
  exchange$: Observable<Exchange>;
  peerTrendsReportWorkbookAsync$: Observable<AsyncStateObj<Workbook>>;
  unsubscribe$ = new Subject<void>();
  enableFileDownloadSecurityWarning$: Observable<boolean>;

  exchangeSubscription: Subscription;
  enableFileDownloadSecurityWarningSubscription: Subscription;

  permissions = Permissions;
  exchangeId: number;
  isSystemExchange = false;
  peerTrendsReportFeatureFlag: RealTimeFlag = { key: FeatureFlags.PeerDashboard_PeerTrendsReport, value: false };
  enableFileDownloadSecurityWarning = false;

  constructor(
    private route: ActivatedRoute,
    private router: Router,
    private store: Store<fromPeerDashboardReducer.State>,
    private featureFlagService: AbstractFeatureFlagService,
    private settingsService: SettingsService
  ) {
    this.sidebarVisible$ = this.store.pipe(select(fromPeerDashboardReducer.getExchangeDashboardSidebarVisible));
    this.mapHasData$ = this.store.pipe(select(fromPeerDashboardReducer.getExchangeDashboardMapHasData));
    this.mapHasDataError$ = this.store.pipe(select(fromPeerDashboardReducer.getExchangeDashboardMapHasDataError));
    this.exchange$ = this.store.pipe(select(fromSharedPeerReducer.getExchange));
    this.uploadOrgDataModalOpen$ = this.store.pipe(select(fromPeerDashboardReducer.getUploadOrgDataModalOpen));
    this.enableFileDownloadSecurityWarning$ = this.settingsService.selectCompanySetting<boolean>(CompanySettingsEnum.FileDownloadSecurityWarning);
    this.peerTrendsReportWorkbookAsync$ = this.store.pipe(select(fromPeerDashboardReducer.getPeerTrendsReportWorkbook));
    this.featureFlagService.bindEnabled(this.peerTrendsReportFeatureFlag, this.unsubscribe$);
  }

  get peerTrendsReportEnabled(): boolean {
    return this.isSystemExchange && !!this.peerTrendsReportFeatureFlag.value;
  }

  manageJobsClick(): void {
    this.router.navigate(['manage'], { relativeTo: this.route.parent.parent });
  }

  referCompanyClick(): void {
    this.store.dispatch(new fromExchangeRequestActions.OpenExchangeRequestModal(ExchangeRequestTypeEnum.ReferPayfactorsCompany));
  }

  mapClick(): void {
    this.router.navigate(['map'], { relativeTo: this.route.parent.parent });
  }

  handlePeerTrendsReportClick(): void {
    this.peerTrendReportModal.open();
  }

  getTitle(hasData: boolean, hasDataError: boolean): string {
    if (hasDataError) { return 'Failed to get map data'; }
    if (!hasData) { return 'No exchange map data available'; }
    return '';
  }

  openUploadOrgDataModal() {
    this.store.dispatch(new fromUploadOrgDataAction.OpenUploadOrgDataModal());
  }

  closeUploadOrgDataModal() {
    this.store.dispatch(new fromUploadOrgDataAction.CloseUploadOrgDataModal());
  }

  handleUploadOrgData(uploadData: any) {
    this.store.dispatch(new fromUploadOrgDataAction.UploadFile(uploadData));
  }

  handleSecurityWarningConfirmed(isConfirmed) {
    if (isConfirmed) {
      this.exportExchangeJobs();
    }
  }

  handleExportExchangeJobs(): void {
    if (this.enableFileDownloadSecurityWarning) {
      this.fileDownloadSecurityWarningModal.open();
    } else {
      this.exportExchangeJobs();
    }
  }

  exportExchangeJobs(): void {
    this.store.dispatch(new fromExchangeDashboardActions.ExportExchangeJobs({exchangeId: this.exchangeId}));
  }

  // Lifecycle
  ngOnInit() {
    this.store.dispatch(new fromCompanyContextActions.GetCompanyContext());

    this.store.dispatch(new fromExchangeDashboardActions.CloseSidebar());

    this.exchangeSubscription = this.exchange$.subscribe(ex => {
        this.store.dispatch(new fromExchangeDashboardActions.LoadMapCount(ex.ExchangeId));
        this.exchangeId = ex.ExchangeId;
        this.isSystemExchange = ex.IsSystemExchange;
    });

    this.enableFileDownloadSecurityWarningSubscription = this.enableFileDownloadSecurityWarning$.subscribe(isEnabled => {
      if (isEnabled) {
        this.enableFileDownloadSecurityWarning = true;
      }
    });
  }

  ngOnDestroy() {
    this.exchangeSubscription.unsubscribe();
    this.enableFileDownloadSecurityWarningSubscription.unsubscribe();
    this.unsubscribe$.next();
  }
}
