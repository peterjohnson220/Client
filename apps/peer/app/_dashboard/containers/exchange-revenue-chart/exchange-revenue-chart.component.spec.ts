import { NO_ERRORS_SCHEMA } from '@angular/core';
import { TestBed, ComponentFixture } from '@angular/core/testing';
import { ActivatedRoute } from '@angular/router';

import { StoreModule, Store, combineReducers } from '@ngrx/store';
import spyOn = jest.spyOn;

import * as fromRootState from 'libs/state/state';
import {  ExchangeChartTypeEnum } from 'libs/models';
import { ActivatedRouteStub } from 'libs/test/activated-route-stub';

import * as fromExchangeDashboardActions from '../../actions/exchange-dashboard.actions';
import * as fromPeerDashboardReducer from '../../reducers';
import { ExchangeRevenueChartComponent } from './exchange-revenue-chart.component';

describe('Peer Dashboard - Revenue Chart', () => {
  let fixture: ComponentFixture<ExchangeRevenueChartComponent>;
  let instance: ExchangeRevenueChartComponent;
  let store: Store<fromPeerDashboardReducer.State>;
  let route: any;

  // Configure Testing Module for before each test
  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        StoreModule.forRoot({
          ...fromRootState.reducers,
          peer_dashboard: combineReducers(fromPeerDashboardReducer.reducers)
        }),
      ],
      providers: [
        {
          provide: ActivatedRoute,
          useValue: new ActivatedRouteStub(),
        }
      ],
      declarations: [
        ExchangeRevenueChartComponent
      ],
      // Shallow Testing
      schemas: [ NO_ERRORS_SCHEMA ]
    });

    store = TestBed.inject(Store);
    route = TestBed.inject(ActivatedRoute);

    route.setParamMap({ id: 1 });

    fixture = TestBed.createComponent(ExchangeRevenueChartComponent);
    instance = fixture.componentInstance;

    spyOn(store, 'dispatch');
  });

  it('should show the revenue chart and title', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should dispatch a LoadDetailChart for the revenue on seriesClick', () => {
    const action = new fromExchangeDashboardActions.LoadDetailChart({
      ExchangeId: 1,
      ChartType: ExchangeChartTypeEnum.Revenue,
      Category: 'Test'
    });

    instance.seriesClick({ category: action.payload.Category });

    expect(store.dispatch).toHaveBeenCalledWith(action);
  });

});
