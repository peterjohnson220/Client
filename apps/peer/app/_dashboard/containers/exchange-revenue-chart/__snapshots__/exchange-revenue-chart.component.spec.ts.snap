// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Peer Dashboard - Revenue Chart should show the revenue chart and title 1`] = `
<pf-exchange-revenue-chart
  __ngContext__={[Function LRootView]}
  exchangeId={[Function Number]}
  loadingRevenueChartItems$={[Function Store]}
  loadingRevenueChartItemsError$={[Function Store]}
  revenueChartItems$={[Function Store]}
  route={[Function ActivatedRouteStub]}
  store={[Function Store]}
>
  <kendo-chart>
    <kendo-chart-legend
      position="bottom"
    />
    <kendo-chart-area
      background="none"
      height="250"
    />
    <kendo-chart-tooltip>
    </kendo-chart-tooltip>
    <kendo-chart-series>
      <kendo-chart-series-item
        categoryfield="Category"
        field="Value"
        type="donut"
      />
    </kendo-chart-series>
  </kendo-chart>
</pf-exchange-revenue-chart>
`;
